### ISSUE TITLE
New lints: an enum variant changes from unit/tuple/struct to another kind. 

### ISSUE THREAD

**Issue by suaviloquence** [obi1kenobi/cargo-semver-checks#884]
### Describe your use case

We have lints for when a `struct` changes "kind": 

- [`tuple_struct_to_plain_struct`](https://github.com/obi1kenobi/cargo-semver-checks/blob/main/src/lints/tuple_struct_to_plain_struct.ron)
- [`unit_struct_changed_kind`](https://github.com/obi1kenobi/cargo-semver-checks/blob/main/src/lints/unit_struct_changed_kind.ron)

However, currently `cargo-semver-checks` does not check when an enum variant changes kind like this.

from: 

```rust
pub enum Enum { 
    WasTuple(),
    WasStruct {},
    WasUnit,
    TupleItem (()),
    StructItem { a: (), },
    Unit,
}
```

to:

```rust
pub enum Enum {
    WasTuple {},
    WasStruct,
    WasUnit(),
    TupleItem { a: () },
    StructItem,
    Unit(()),
}
```

the first three variants would break with:

```rust
let _ = Enum::WasTuple;
// since it's now a tuple variant, this still gives us a value,
// but it's a `fn(()) -> Enum` tuple constructor, not an `Enum` instance.
let _: Enum = Enum::WasStruct;
let _ = Enum::WasUnit;
```

the second three also have breakage in a similar fashion, and `cargo-semver-checks` currently does not catch the
fields_missing/removed lints because these variants changed kind

### Describe the solution you'd like

Thus, it would be helpful to have lint(s) that detect this change for **constructible** enum variants - if a tuple/struct variant is marked `#[non_exhaustive]` in the baseline version, this is not a breaking change because it was not possible to construct these variants in the baseline version.
